optaplanner-examples/src/main/java/org/optaplanner/examples/curriculumcourse/domain/Lecture.java

```diff
@@ -60,7 +60,8 @@ public boolean isPinned() {
         return pinned;
     }
 
-    public void setPinned(boolean pinned) {
+    // Object type Boolean required due too Lambda creating on JDK 11
+    public void setPinned(Boolean pinned) {
```

Comment:
That's a pain - users will want to use a boolean instead of a Boolean here.

Looks like a regression in JDK 11 for LamdbaMetaFactory. We must create a ticket for that.

Reply:
@rsynek Would you mind creating a new git repo with a reproducer that succeeds in JDK 8 but not in JDK 11?
Basically it's a copy paste of this file:
https://github.com/ge0ffrey/ge0ffrey-presentations/blob/master/code/fasterreflection/fasterreflection-framework/src/main/java/be/ge0ffrey/presentations/fasterreflection/framework/LambdaMetafactoryBeanPropertyReader.java
with
```
private final BeanPropertyReader lambdaMetafactoryBeanPropertyReader = new LambdaMetafactoryBeanPropertyReader(Foo.class, "pinned");
lambdaMetafactoryBeanPropertyReader.executeGetter(foo);
```

Once we have such a bug report, let's mail it to Andrew and ask what's the best way forward to bring them to the openJDK's team's attention.

Reply:
@ge0ffrey It's not a regression - they just deliberately added type checking. I think we can work around this on our end by creating extra methods that process primitive types and not just `Object`.

Reply:
@ge0ffrey +1, changing primitive types to Object type is backward incompatible workaround, not a solution.  @triceo thanks for making me realize that.

Given BiConsumer does not accept primitive types, we may need to map to BiConsumers for every primitive type, something like https://docs.oracle.com/javase/8/docs/api/java/util/function/ObjIntConsumer.html

Reply:
@rsynek @triceo Agreed, please create a jira for this work.
It will involve fixing this the LambdaMetafactory.metafactory() calls in this class:
https://github.com/kiegroup/optaplanner/blob/5c33d3c0490b5d42a40178b76003874a363b720b/optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/LambdaBeanPropertyMemberAccessor.java#L87

That's 5 lines of headache code. The first thing to know that is any set of method parameters, the first type is the return type. Not that the JDK javadocs mention that anywhere...